app/services/list_publisher.rb

```diff
@@ -0,0 +1,13 @@
+class ListPublisher
+  attr_reader :tag
+
+  def initialize(tag)
+    @tag = tag
+  end
+
+  def perform
+    groups = TopicPresenter.new(tag).build_groups
```

Comment:
This needs to use `TagPresenter.for(tag)` so that it will also support mainstream browse pages.